Cpp_test/lib/ipc-shm-receive.cpp

```diff
@@ -0,0 +1,110 @@
+#include "ipc-shm-receive.h"
+
+IPCShmReceive::~IPCShmReceive()
+{
+
+	shm_unlink(this->info.method_name);
+	if (pthread_cond_destroy(&(this->shm_ptr->cond_re)) != 0)
+	{
+		std::cerr << "Error at pthread_cond_destroy(): receive" << std::endl;
+	}
+	if (pthread_cond_destroy(&(this->shm_ptr->cond)) != 0)
+	{
+		std::cerr << "Error at pthread_cond_destroy(): send" << std::endl;
+	}
+	if (pthread_mutex_destroy(&(this->shm_ptr->mutex)) != 0)
+	{
+		std::cerr << "Error at pthread_mutex_destroy()" << std::endl;
+	}
+}
+
+void IPCShmReceive::init()
+{
+	this->file_handler.open_file();
+	this->open_shm();
+	this->map_shm();
+	if (this->shmd > 0)
+	{
+		close(this->shmd);
+	}
+}
+
+void IPCShmReceive::transfer()
+{
+	std::cout << "Receiving..." << std::endl;
+	unsigned long total_received_bytes{0};
+
+	while (!this->is_end && this->shm_ptr->is_init)
+	{
+		lock_mutex();
+		while (this->shm_ptr->data_version_received == this->shm_ptr->data_version)
+		{
+			if (pthread_cond_wait(&(this->shm_ptr->cond), &(this->shm_ptr->mutex)) != 0)
+			{
+				throw std::runtime_error(std::string("ERROR: pthread_cond_wait() send: ") + strerror(errno));
+			}
+		}
+
+		if (this->shm_ptr->shared_mem_size != this->shm_size_in_bytes)
+		{
+			pthread_mutex_destroy(&(this->shm_ptr->mutex));
```

Comment:
You should remove it because throwing here will call the destructor (if you catch the throw somewhere in the program